### PR TITLE
Stability fixes and test coverage improvements

### PR DESCRIPTION
**Pull request by BrianFarnhill**

### CODE REVIEW

**Comment by BrianFarnhill** [comment]
@caadam Can a brother get a code review?

File: Modules/xSharePoint/Modules/xSharePoint.Util/xSharePoint.Util.psm1 [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
-        [System.Management.Automation.PSCredential]
-        $Credential,
+        [parameter(Mandatory = $true,Position=1)] [string] $UserName
+    )
 
-        [parameter(Mandatory = $false)]
-        [HashTable]
-        $Arguments,
+    $domainName = $UserName.Split('\')[0]
+    $accountName = $UserName.Split('\')[1]
```
Comment (caadam) [inline Modules/xSharePoint/Modules/xSharePoint.Util/xSharePoint.Util.psm1]: If you are going to do string splitting you need to be defensive with your treatment of the $userName varaible. Either you need to do a test for \ here, or ideally, validate the variable at the start of the DSC to ensure it is a match for DOMAIN\USER

**Comment by caadam** [comment]
Happy with changes. See inline comment for only recommendation.

**Comment by BrianFarnhill** [comment]
Thanks cam, feedback has been addressed - merging this in to dev

**Pull request merged**